src/stylesheets/core/_functions.scss

```diff
@@ -900,28 +900,61 @@ Derive a color from a color shortcode
 ----------------------------------------
 */
 
-@function color($shortcode) {
-  $shortcode: smart-quote(unpack($shortcode));
-  @if not $shortcode {
-    @return false;
+@function color($value, $flags...) {
+  $value: unpack($value);
+
+  // Non-token colors may be passed with specific flags
+  @if type-of($value) == color {
+    // override or set-theme will allow any color
+    @if index($flags, override) or index($flags, set-theme) {
+      // override + no-warn will skip warnings
+      @if index($flags, no-warn) {
+        @return $value;
+      }
+
+      @if $theme-show-compile-warnings {
+        @warn 'Override: `#{$value}` is not a USWDS color token.';
+      }
+
+      @return $value;
+    }
```

Comment:
The color function can now include a number of flags like `color(value, flag-1, flag-2, ...)`. `override` and `set-theme` flags allow passing a color. `no-warn` suppresses the warning.